### PR TITLE
Allow structured outputs via function calling

### PR DESCRIPTION
**Pull request by greysteil**
**Describe the change**

Allows `strict: true` to be passed to OpenAI within a function's definition, triggering OpenAI's new [structured outputs](https://openai.com/index/introducing-structured-outputs-in-the-api/) setup to ensure the function's schema is strictly adhered to.

**Provide OpenAI documentation link**

https://openai.com/index/introducing-structured-outputs-in-the-api/

**Describe your solution**

Add `Strict` to the properties of `FunctionDefinition`. Only serialise it in the JSON if its value is `true` (to avoid wasting tokens).

**Tests**
Added a test.

**Additional context**
None

### CODE REVIEW

**Comment by greysteil** [comment]
@sashabaranov the linting error on my first commit wasn't related to this PR. I've fixed it in the second commit since it was a trivial one.

**Comment by sashabaranov** [comment]
@greysteil thank you for the PR! Could you please fix the merge conflict and add an integration test — it's pretty hard to follow all the OpenAI API updates these days

**Comment by greysteil** [comment]
@sashabaranov done!

**Review by sashabaranov — APPROVED** [review]
